ironfish/src/primitives/target.test.ts

```diff
@@ -64,81 +65,73 @@ describe('Target', () => {
 })
 
 describe('Calculate target', () => {
-  it('can increase target (which decreases difficulty) if its taking too long to mine a block (20+ seconds since last block)', () => {
+  it('increases difficulty if a new block is coming in before the target range time', () => {
     const now = new Date()
-    // for any time 20-29 seconds after the last block, difficulty should decrease by previous block's difficulty / BigInt(2048)
-    for (let i = 1; i < 10; i++) {
-      const time = new Date(now.getTime() + 20000 + i * 1000)
+    // for any time 0 - 55 seconds after the last block, difficulty should increase by previous block's difficulty / BigInt(2048) * m
+    for (let i = 0; i < 55; i++) {
+      const time = new Date(now.getTime() + i * 1000)
 
       const difficulty = BigInt(231072)
       const target = Target.fromDifficulty(difficulty)
 
-      const diffInDifficulty = difficulty / BigInt(2048)
-
-      const newDifficulty = Target.calculateDifficulty(time, now, difficulty)
-      const newTarget = Target.calculateTarget(time, now, target)
-
-      expect(newDifficulty).toBeLessThan(difficulty)
-      expect(BigInt(newDifficulty) + diffInDifficulty).toEqual(difficulty)
-
-      expect(newTarget.asBigInt()).toBeGreaterThan(target.asBigInt())
-    }
+      const bucketFromParent =
```

Comment:
can we extract this to a method that all these tests can use? otherwise it's a lot of copy/paste code that's part of the test setup